frontend/public/components/monitoring/alerting.tsx

```diff
@@ -1764,6 +1825,20 @@ const RulesPage_: React.FC<{}> = () => {
     type: 'name',
   } as RowFilter;
 
+  const ruleAdditionalSources = React.useMemo(() => {
+    if (data) {
+      const additionalSources = new Set<string>();
+      data.forEach((rule) => {
+        const source = alertingRuleSource(rule);
+        if (source !== AlertSource.Platform && source !== AlertSource.User) {
+          additionalSources.add(source);
+        }
+      });
+      return Array.from(additionalSources).map((item) => ({ id: item, title: _.startCase(item) }));
+    }
+    return [];
+  }, [data]);
+
```

Comment:
I notice that this code is reapeated twice. It's probably complex enough to be worth creating a helper function to avoid the repetition.

Reply:
I refactored it as a function